Tests/SwiftDocCTests/Diagnostics/DiagnosticTests.swift

```diff
@@ -140,6 +141,27 @@ class DiagnosticTests: XCTestCase {
         \(expectedNoteLocation): note: The message of the note.
         """)
     }
+
+    func testDoxygenDiagnostic() throws {
+        let tempURL = try createTemporaryDirectory()
+
+        let bundleURL = try Folder(name: "InheritedDocs.docc", content: [
+            InfoPlist(displayName: "Inheritance", identifier: "com.test.inheritance"),
+            CopyOfFile(original: Bundle.module.url(
+                        forResource: "TestDoxygenDiagnostics.symbols", withExtension: "json",
+                        subdirectory: "Test Resources")!),
+        ]).write(inside: tempURL)
+        let (_, bundle, _) = try loadBundle(from: bundleURL)
+
+        let reference = ResolvedTopicReference(bundleIdentifier: bundle.identifier, path: "/documentation/Test/Test/HTMLStringWithMarkdown", sourceLanguage: .objectiveC)
+
+        let node = DocumentationNode(reference: reference, kind: .typeMethod, sourceLanguage: .objectiveC, name: .conceptual(title: name), markup: Document(parsing: "# \(name)"), semantic: nil)
```

Comment:
I may be missing something but this doesn't _look_ like it's actually loading a symbol from the symbol graph to me. Does this test reproduce the issue when you remove the `if !allKnownDirectiveName.contains(comment.name) {` fix?

Reply:
I agree testing a symbol graph makes more sense. I updated the test with both the scenarios when there is a known directive and when there is none. Please take a look.